src/mon/PGMap.cc

```diff
@@ -2875,6 +2875,25 @@ void PGMap::get_health_checks(
       d.detail.swap(detail);
     }
   }
+
+  // SNAPTRIMQ_SLOW
+  if (!pg_stat.empty() && cct->_conf->mon_osd_snap_trim_queue_warn_on > 0) {
+    uint32_t snapthreshold = cct->_conf->mon_osd_snap_trim_queue_warn_on;
+    uint64_t snaptrimq_exceeded = 0;
+    for (auto& i: pg_stat) {
+      if (i.second.snaptrimq_len >= snapthreshold) {
+        snaptrimq_exceeded++;
+      }
+    }
+    if (snaptrimq_exceeded) {
+      list<string> detail;
+      stringstream ss;
+      ss << "snap trim queue for " << snaptrimq_exceeded << " pg(s) hit the " << snapthreshold << " point. ";
```

Comment:
+      ss << "snap trim queue for " << snaptrimq_exceeded << " pg(s) >= " << snapthreshold << " (mon_osd_snap_trim_queue_warn_on)";